diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt

```diff
@@ -58,11 +54,11 @@ class DiktatGradlePluginFunctionalTest {
 
         val diktatCheckBuildResult = result.task(":$DIKTAT_CHECK_TASK")
         requireNotNull(diktatCheckBuildResult)
-        Assertions.assertTrue(testProjectDir.root.walkTopDown().filter { it.name == "test.txt" }.firstOrNull() != null)
         Assertions.assertEquals(TaskOutcome.FAILED, diktatCheckBuildResult.outcome)
-        val file = testProjectDir.root.walkTopDown().filter { it.name == "test.txt" }.first()
+        val file = testProjectDir.root.walkTopDown().filter { it.name == "test.txt" }.firstOrNull()
+        Assertions.assertTrue(file != null)
```

Comment:
Does `kotlin.test.assertNotNull` work in this case? It should have contract to enable smart cast of `file` to non-nullable type.